File: road/codec/consts.go
package codec

/********************************************************************
created:    2022-09-03
author:     lixianmin

Copyright (C) - All Rights Reserved
*********************************************************************/

// Codec constants.
const (
	HeaderSize    = 4
	MaxPacketSize = 1 << 24 //16MB
)

// client handshake → server handshake → client handshakeAck → server heartbeat
const (
	Handshake    = 0x01 // Handshake represents a handshake: request(client) <====> handshake response(server)
	HandshakeAck = 0x02 // HandshakeAck represents a handshake ack from client to server
	Heartbeat    = 0x03 // Heartbeat represents a heartbeat
	Data         = 0x04 // Data represents a common data packet
	Kick         = 0x05 // Kick represents a kick-off packet, disconnect message from server
)

File: web/src/code/serde/consts.js
'use strict'

/********************************************************************
 created:    2023-06-09
 author:     lixianmin

 Copyright (C) - All Rights Reserved
 *********************************************************************/

export const PacketKind = {
    Handshake: 1,       // 连接建立后, 服务器主动发送handshake
    Heartbeat: 2,       // client定期发送心跳
    Kick: 3,            // server踢人
    UserDefined: 1000,  // 用户自定义的类型, 从这里开始
}

